prymer/primer3/primer3_input.py

```diff
@@ -81,28 +87,48 @@
 PRIMER_WT_TM_GT -> 1.0
 """
 
+from dataclasses import MISSING
 from dataclasses import dataclass
+from dataclasses import fields
 from typing import Any
 from typing import Optional
 
 from prymer.api.span import Span
 from prymer.primer3.primer3_input_tag import Primer3InputTag
-from prymer.primer3.primer3_parameters import Primer3Parameters
+from prymer.primer3.primer3_parameters import PrimerAndAmpliconParameters
+from prymer.primer3.primer3_parameters import ProbeParameters
 from prymer.primer3.primer3_task import Primer3TaskType
 from prymer.primer3.primer3_weights import PrimerAndAmpliconWeights
 from prymer.primer3.primer3_weights import ProbeWeights
 
 
 @dataclass(frozen=True, init=True, slots=True)
 class Primer3Input:
-    """Assembles necessary inputs for Primer3 to orchestrate primer and/or primer pair design."""
+    """Assembles necessary inputs for Primer3 to orchestrate primer, primer pair, and/or internal
+    probe design."""
 
     target: Span
     task: Primer3TaskType
-    params: Primer3Parameters
-    primer_weights: Optional[PrimerAndAmpliconWeights] = PrimerAndAmpliconWeights()
+    primer_and_amplicon_params: Optional[PrimerAndAmpliconParameters] = None
+    probe_params: Optional[ProbeParameters] = None
+    primer_weights: Optional[PrimerAndAmpliconWeights] = None
     probe_weights: Optional[ProbeWeights] = None
 
+    def __post_init__(self) -> None:
+        # check for at least one set of params
+        # for the set of params given, check that weights were given; use defaults if not given
+        if self.primer_and_amplicon_params is None and self.probe_params is None:
+            raise ValueError(
+                "Primer3 requires at least one set of parameters"
+                " for either primer or probe design"
+            )
+
+        if self.primer_and_amplicon_params is not None and self.primer_weights is None:
+            object.__setattr__(self, "primer_weights", PrimerAndAmpliconWeights())
+
+        if self.probe_params is not None and self.probe_weights is None:
+            object.__setattr__(self, "probe_weights", ProbeWeights())
```

Comment:
**suggestion** I think it'd be appropriate to make each of these the default value for the attribute, and change the type of the attributes to not be `Optional`